internal/core/src/segcore/SegmentInterface.cpp

```diff
@@ -178,32 +178,72 @@ SegmentInternalInterface::get_real_count() const {
     return res->fields_data()[0].scalars().long_data().data(0);
 }
 
+/**
+ * Find the first timestamp greater than the target and the return its' index in the timestamps.
+ * TODO: move to a proper place.
+ *
+ * @param timestamps
+ * @param start
+ * @param end
+ * @param target
+ * @return The index of answer, -1 will be returned if no timestamp is bigger than the target.
+ */
+int64_t
+find_first_bigger(const ConcurrentVector<Timestamp>& timestamps,
```

Comment:
how about change it to :
```
int64_t
upper_bound(const ConcurrentVector<Timestamp>& timestamps,
                      int64_t start,
                      int64_t end,
                      Timestamp target);
 ```         
 
Returns an index pointing to the first element in the range [`start`, `end`) such that value < `target`  or `end` if no such element is found

Reply:
https://en.cppreference.com/w/cpp/algorithm/upper_bound